service/matching/taskListManager.go

```diff
@@ -170,6 +179,27 @@ func (c *taskListManagerImpl) AddTask(execution *s.WorkflowExecution, taskInfo *
 	return err
 }
 
+func (c *taskListManagerImpl) SyncMatchQueryTask(ctx context.Context, queryTask *queryTaskInfo) error {
+	c.startWG.Wait()
+
+	domainID := queryTask.queryRequest.GetDomainUUID()
+	we := queryTask.queryRequest.QueryRequest.Execution
+	taskInfo := &persistence.TaskInfo{
+		DomainID:   domainID,
+		RunID:      we.GetRunId(),
+		WorkflowID: we.GetWorkflowId(),
+	}
+
+	request := &getTaskResult{task: taskInfo, C: make(chan *syncMatchResponse, 1), queryTask: queryTask}
```

Comment:
Why can't we use the existing trySyncMatch?

Reply:
the existing sync match won't block, and if there is no poller, existing sync match would write to db. in this query case, we want block on sync match, and not write to db.